apps/middleman-workflows/src/lib/dal/transaction.ts

```diff
@@ -0,0 +1,40 @@
+import { db } from "../db";
+import { Transaction, transactionsTable } from "../db/schema";
+import { eq } from "drizzle-orm";
+
+export async function getTransaction(transactionId: number) {
+  return await db.query.transactionsTable.findFirst({
+    where: eq(transactionsTable.id, transactionId),
+    with: { dependsOn: true },
+  });
+}
+
+export async function getDependantTransactions(transactionId: number) {
+  return await db.query.transactionsTable.findMany({
+    where: eq(transactionsTable.dependsOn, transactionId),
+  });
+}
+
+export async function createTransaction(transaction: Transaction) {
+  return await db
+    .insert(transactionsTable)
+    .values(transaction)
+    .returning()
+    .then((res) => res[0]);
+}
```

Comment:
Similar to activities there is no use for this either.  Automatically creating transactions.

```suggestion
```